src/web/auth/jwt.rs

```diff
@@ -53,13 +53,14 @@ struct Claims {
 
 impl Jwks {
     /// Returns a Jwks retrieved from the location identified by the given URI.
-    pub fn from_uri(uri: &str) -> Result<Self> {
-        let load_err = Error::Configuration(format!("Can't load Jwks state from {}", uri));
+    pub async fn from_uri(uri: &str) -> Result<Self> {
+        let load_err = || Error::Configuration(format!("Can't load Jwks state from {}", uri));
```

Comment:
Why is it callable and not a value?

Reply:
Oh, it's the `async` thing :( I need to handle an error case twice:

* once after `reqwest::get(uri).await`
* and then right after `response.text().await`

In the sync version we could combine the two using `and_then()`, but now we need `.await` in between, which messes that up. Because I use `load_err` twice, rustc complains that the value has already been moved

```
error[E0382]: use of moved value: `load_err`
  --> src/web/auth/jwt.rs:61:47
   |
57 |         let load_err = Error::Configuration(format!("Can't load Jwks state from {}", uri));
   |             -------- move occurs because `load_err` has type `auth::Error`, which does not implement the `Copy` trait
...
60 |                 let response = reqwest::get(uri).await.map_err(|_| load_err)?;
   |                                                                --- -------- variable moved due to use in closure
   |                                                                |
   |                                                                value moved into closure here
61 |                 response.text().await.map_err(|_| load_err)?
   |                                               ^^^ -------- use occurs due to use in closure
   |                                               |
   |                                               value used here after move
```

Reply:
Omg